src/stub/core_impl/register_info.rs

```diff
@@ -0,0 +1,134 @@
+use super::prelude::*;
+use crate::protocol::commands::ext::RegisterInfo;
+
+use crate::arch::lldb::{Encoding, Format, Generic, Register, RegisterInfo as LLDBRegisterInfo};
+use crate::arch::Arch;
+
+impl<T: Target, C: Connection> GdbStubImpl<T, C> {
+    pub(crate) fn handle_register_info(
+        &mut self,
+        res: &mut ResponseWriter<'_, C>,
+        target: &mut T,
+        command: RegisterInfo,
+    ) -> Result<HandlerStatus, Error<T::Error, C::Error>> {
+        if !target.use_register_info() {
```

Comment:
I've got a sneaking suspicion that this dead-code-elimination guard might need to be "beefed up" a bit to make sure we aren't introducing undue binary overhead to targets not using register info...

Can you double check and make sure that `./example_no_std/check_size.sh` returns the same value before/after this PR?

Reply:
Testing off commit 1fde6f7
before this PR
```
section               size      addr
...
.text                15010      1792
...
Total                19524
```
after this PR
```
section               size      addr
...
.text                14994      1792
...
Total                19508
```

Reply:
huh, wasn't expecting a shrink, but hey, I'll take it! the LLVM optimizer works in mysterious ways...